common/src/main/java/org/samo_lego/taterzens/commands/NpcCommand.java

```diff
@@ -198,6 +215,106 @@ private static int selectTaterzenById(CommandContext<CommandSourceStack> context
         return 1;
     }
 
+    private static int selectTaterzenByName(CommandContext<CommandSourceStack> context) throws CommandSyntaxException {
+        CommandSourceStack source = context.getSource();
+        String name = StringArgumentType.getString(context, "name");
+
+        // first count how many NPCs have the same name.
+        // If there is more than one NPC with an identical name, the command will fail.
+        // Otherwise, the NPC will be selected.
+        // In case no NPC with that name is found, taterzen is null
+        TaterzenNPC taterzen = null;
+        int count = 0; // number of npcs with identical name
+        for (TaterzenNPC npcIt : TATERZEN_NPCS) {
+            if (npcIt.getName().getString().equals(name)) {
+                taterzen = npcIt;
+                count++;
```

Comment:
Then you could also use `boolean found` instead of `int count`

Reply:
Good idea for saving loop iterations! I made changes according to your suggestion. But I did not use a boolean, since using the count variable (which can become greater than 1 and, theoretically, take even negative values) makes such a boolean redundant.